### DISCUSSION TITLE
Blackmagic Webpresenter as SRT Source for ReStreamer

### DISCUSSION THREAD

**Discussion by streamsolutions** [datarhei/restreamer#656] [Q&A]
Did anyone was lucky creating a Config for the Blackmagic Webpresenter so data is send to ReStreamer via SRT?

I was trying something like this as xml configuration but no luck probing failed ... 

```
<?xml version="1.0" encoding="UTF-8" ?>
<streaming>
    <service>
        <name>Why is this not working</name>
        <servers>
            <server>
                <name>WebPresenter SRT</name>
                <url>srt://myServerIP:6000</url>
                <srt-extensions>
                    <stream-id>
                        <item key="mode" value="caller"/>
                        <item key="transtype" value="live"/>
                        <item key="streamid" value="1f23e9f0-7897-4dde-8e05-4bea93a6d309,mode:publish" />
                    </stream-id>
                </srt-extensions>
            </server>
        </servers>
        <profiles default="Streaming High">
            <profile>
                <name>Streaming High</name>
                <!-- Define <config> elements to specify target bitrates for Streaming High profile -->
            </profile>
            <!-- Streaming quality settings-->
        </profiles>
    </service>
</streaming>
```

the webpresenter seams to send additional data that might hinder the correct recognition?!?

```
bmd_uuid=xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx,
bmd_name=Blackmagic Web Presenter HD
```

**Comment by jorblad** [comment]
I also have that problem I think I have tried to many different variants but have not found a way to make it work.

**Comment by ioppermann** [comment]
Disclaimer: I don't have access to a Webpresenter and trying to figure out what is going on by searching for some information.

The SRT streamid configuration in Webpresenter seems to be a bit tricky, and it looks like that it has it's own notion of how a streamid should like: `!#::r=some_resource,m=publish` (based on https://forum.blackmagicdesign.com/viewtopic.php?f=4&t=188339):

```
<srt-extensions>
    <stream-id>
        <item key=”r” value=”some_resource”/>
        <item key=”m” value=”publish”/>
    </stream-id>
</srt-extensions>
```

However, this is not what Restreamer supports.

As of the latest version it possible to send a stream without a streamid, so the configuration will simply look like:

```
<server>
    <name>WebPresenter SRT</name>
    <url>srt://myServerIP:6000</url>
</server>
```

You can give this a try. In Restreamer in the "Advanced settings" you can select this stream in "Receive mode".

**Comment by shrippen** [comment]
Sadly I am not able to get this working. Has anyone got a working config?
